kong/resty/dns/client.lua

```diff
@@ -747,41 +748,15 @@ local queue = setmetatable({}, {__mode = "v"})
 local function executeQuery(premature, item)
   if premature then return end
 
-  local r, err = resolver:new(config)
-  if not r then
-    item.result, item.err = r, "failed to create a resolver: " .. err
-  else
-    --[[
-    log(DEBUG, PREFIX, "Query executing: ", item.qname, ":", item.r_opts.qtype, " ", fquery(item))
-    --]]
-    add_status_to_try_list(item.try_list, "querying")
-    item.result, item.err = r:query(item.qname, item.r_opts)
-    if item.result then
-      --[[
-      log(DEBUG, PREFIX, "Query answer: ", item.qname, ":", item.r_opts.qtype, " ", fquery(item),
-              " ", frecord(item.result))
-      --]]
-      parseAnswer(item.qname, item.r_opts.qtype, item.result, item.try_list)
-      --[[
-      log(DEBUG, PREFIX, "Query parsed answer: ", item.qname, ":", item.r_opts.qtype, " ", fquery(item),
-              " ", frecord(item.result))
-    else
-      log(DEBUG, PREFIX, "Query error: ", item.qname, ":", item.r_opts.qtype, " err=", tostring(err))
-      --]]
-    end
-  end
+  item.result, item.err = individualQuery(item.qname, item.r_opts, item.try_list)
 
   -- query done, but by now many others might be waiting for our result.
   -- 1) stop new ones from adding to our lock/semaphore
   queue[item.key] = nil
   -- 2) release all waiting threads
   item.semaphore:post(math_max(item.semaphore:count() * -1, 1))
   item.semaphore = nil
-  ngx.sleep(0)
-  -- 3) destroy the resolver -- ditto in individualQuery
-  if r then
-    r:destroy()
-  end
```

Comment:
After the change the `destroy()` happens at a different time (before the yield). Do we have some reason to destroy after the yield in the original code?

Reply:
carefully examined the logic and referred to the official resolver library's examples. Here, we can proactively destroy 'r' since we've already extracted the data from it.